controllers/core/flagd/resources/deployment.go

```diff
@@ -110,14 +111,7 @@ func (r *FlagdDeployment) GetResource(ctx context.Context, flagd *api.Flagd) (cl
 
 	// override settings for the injected container for flagd standalone deployment mode
 	deployment.Spec.Template.Spec.ImagePullSecrets = imagePullSecrets
-	if len(r.FlagdConfig.Labels) > 0 {
-		maps.Copy(deployment.Spec.Template.ObjectMeta.Labels, r.FlagdConfig.Labels)
-	}
-	if len(r.FlagdConfig.Annotations) > 0 {
-		maps.Copy(deployment.Spec.Template.ObjectMeta.Annotations, r.FlagdConfig.Annotations)
-	}
```

Comment:
With the change above, I don't think we need to do this map copy; we defined `deployment.Spec.Template.ObjectMeta.Annotations` based on the contents of `r.FlagdConfig.Annotation` above.

I tested this locally and the injected pod's annotations look fine.

Reply:
Yeah, makes sense. I think when I originally coded this I was following a rote algorithm: "Merge labels anywhere there is an image pull secret."

LGTM